### DISCUSSION TITLE
Is it possible to multi-target wix4 projects?

### DISCUSSION THREAD

**Discussion by jameswtelfer** [wixtoolset/issues#7555] [Questions]
I am looking at porting an existing .NET Framework application installer from 32-bit to 64-bit.  The existing installer is built using wix3 to generate a number of merge modules which are then glued together into an MSI and bundle installer.

For business reasons, we want to keep the existing 32-bit installer, while also generating a new 64-bit installer.  For 90% of our merge modules, the files within the two will be identical; MSIL assemblies targeting Any CPU.

The application is made up from a (large) number of VS solutions, each containing a number of .NET projects and normally a single wixproj that builds the MSM.  We want to keep that general pattern.

Is it possible, in wix4, to create a single wixproj that would build both a 32-bit and 64-bit MSM from the same set of inputs?  I know I can define different solution Platform entries to build the two separately, but what I'd really like is to build both from a single solution Platform.  My driving use-case here is our MSBuild CI which expects to build each solution only once in its current form.

**Comment by chrpai** [comment]
Yes, No, Yes. :)

The first yes is a single set of .wixproj/.wxs can be used to build x86 and x64 and now ARM64 MSM/MSI.

No, you can't build it once and get multiple MSI types.

however....

Yes, you could have 2 different sln files containing the same projects one build x86 one build x64.  This would maintain the 1:1 your current CI system expects.

I've spent a lot of time in this area if you'd like a demo of my FOSS and how it  does exactly what you talk about.  Checkout https://github.com/iswix-llc/iswix and https://github.com/iswix-llc/iswix-tutorials.   The tool has project templates for both v3 and v4 and works alongside Heatwave to support upgrading v3 projects to v4 and put them in a state where they support the build capabilities above that I mentioned.
